### PR TITLE
Switch db to postgres

### PR DESCRIPTION
**Pull request by MichaelDimitras**
@rthai I refactored the code to use Postgres instead of Mongo and set up Artillery and New Relic

### CODE REVIEW

File: artillery.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      arrivalRate: 100
+    - duration: 90
+      arrivalRate: 250
+    - duration: 90
+      arrivalRate: 500
+    - duration: 90
+      arrivalRate: 1000
+    - duration: 90
+      arrivalRate: 1250
+  variables:
```
Comment (rthai) [inline artillery.yml:18]: Can use a script for this if you choose to refactor.

File: tests/db_tests/postgres.test.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  } else if (data) {
+    queryData = data;
+  }
+}
+
+describe('getRestaurant', () => {
+  
+  it('invokes the callback with data for a successful query', () => {
+    const goodQuery = db.getRestaurant(1, queryCallback);
+    expect(queryData).toBeDefined
```
Comment (rthai) [inline tests/db_tests/postgres.test.js]: toBeDefined should be toBeDefined()

Comment (rthai) [reply tests/db_tests/postgres.test.js]: Your test looks like postgres must be running in order for it to pass? Better to mock the db.

File: tests/db_tests/postgres.test.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  })
+
+  queryData = undefined;
+  
+  it('invokes the callback with data for a successful query', () => {
+    const goodQuery = db.getRestaurant(1, queryCallback);
+    expect(isErr).toBe(true);
```
Comment (rthai) [inline tests/db_tests/postgres.test.js]: Test name should be changed because it looks like you're trying to test a bad query.
How are you getting an error message to set isErr to true? Isn't id: 1 in your db?

**Review by rthai — CHANGES_REQUESTED** [review]
I have some questions about your tests. Please see comments.

**Comment by MichaelDimitras** [comment]
@rthai I fixed the database tests. I will research how to refactor my artillery.yml for future tests.

**Comment by MichaelDimitras** [comment]
@rthai Everything is passing now. Thanks for helping me out with this test!

File: tests/db_tests/postgres.test.js [hunk 4]
```diff
@@ -19,6 +19,7 @@ describe('getRestaurant', () => {
   })
 
   queryData = undefined;
+  isErr = false;
```
Comment (rthai) [inline tests/db_tests/postgres.test.js]: Why is this necessary, if default is false?

File: tests/db_tests/postgres.test.js [hunk 5]
```diff
@@ -19,6 +19,7 @@ describe('getRestaurant', () => {
   })
```
Comment (rthai) [inline tests/db_tests/postgres.test.js]: .toBeDefined()

File: database/postgres.js [hunk 6]
```diff
@@ -10,19 +10,10 @@ const config = {
...
-//       cb(null, data.rows[0]);
-//     }
-//   });
-// };
-
-
-const getRestaurant = (id) => (
+const getRestaurant = (id, cb) => (
   pool.query(`SELECT * FROM restaurants WHERE id=${id}`)
+  .then(res => cb(null, res.rows[0]))
```
Comment (rthai) [inline database/postgres.js]: Sort of weird to be using cb with promises.

Comment (rthai) [reply database/postgres.js]: `const getRestaurant = (id) => (
   pool.query(`SELECT * FROM restaurants WHERE id=${id}`)`

in your server.js
`Places.getRestaurant(req.params.id)
.then(result => response.send(result.rows[0]))
.catch(error => response.status(500).send(error));`
